src/ServerTelemetryChannel/Implementation/ApplicationFolderProvider.cs

```diff
@@ -11,12 +11,15 @@
     using System.Security.Cryptography;
     using System.Security.Principal;
     using System.Text;
+    using Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel.Shared.Implementation;    
 
     internal class ApplicationFolderProvider : IApplicationFolderProvider
     {
+        internal Func<DirectoryInfo, bool> ApplySecurityToDirectory;
+
         private readonly IDictionary environment;
         private readonly string customFolderName;
-        private readonly WindowsIdentity currentIdentity = WindowsIdentity.GetCurrent();
+        private IIdentityProvider identityProvider;
```

Comment:
it can still be readonly. It's initialized in constructor

Reply:
done